web_src/js/components/RepoBranchTagDropdown.js

```diff
@@ -3,43 +3,39 @@ import $ from 'jquery';
 import {vueDelimiters} from './VueComponentLoader.js';
 
 export function initRepoBranchTagDropdown(selector) {
-  $(selector).each(function () {
-    const $dropdown = $(this);
-    const $data = $dropdown.find('.data');
+  $(selector).each(function (dropdownIndex, elRoot) {
     const data = {
+      csrfToken: window.config.csrfToken,
       items: [],
-      mode: $data.data('mode'),
       searchTerm: '',
-      refName: '',
-      noResults: '',
-      canCreateBranch: false,
       menuVisible: false,
       createTag: false,
+      release: null,
+
       isViewTag: false,
       isViewBranch: false,
       isViewTree: false,
+
       active: 0,
-      branchForm: '',
-      branchURLPrefix: '',
-      branchURLSuffix: '',
-      tagURLPrefix: '',
-      tagURLSuffix: '',
-      setAction: false,
-      submitForm: false,
+
+      ...window.config.pageData.branchDropdownDataList[dropdownIndex],
     };
-    $data.find('.item').each(function () {
-      data.items.push({
-        name: $(this).text(),
-        url: $(this).data('url'),
-        branch: $(this).hasClass('branch'),
-        tag: $(this).hasClass('tag'),
-        selected: $(this).hasClass('selected')
-      });
-    });
-    $data.remove();
 
-    // eslint-disable-next-line unicorn/no-this-assignment
-    const elRoot = this;
+    if (data.showBranchesInDropdown) {
+      for (const branch of data.branches) {
+        data.items.push({name: branch, url: branch, branch: true, tag: false, selected: branch === data.defaultBranch});
+      }
+    }
+    if (!data.noTag) {
```

Comment:
I got this error:
<img width="488" alt="截屏2023-03-10 14 48 05" src="https://user-images.githubusercontent.com/17645053/224243920-88029dc6-f328-49ac-a2bd-7f5964087f30.png">

Looks like `data.noTag` is null because `noTag` is not passed to the template. And here the data.noTag is a string, so assume "true" or "false" is pasded to the template, the if always evaluate to true because it is a string and it has length. Maybe should do something like `data.noTag === "false"`

Reply:
Oh, bug.

I believe there was misuse between string `"true"/"false"` and bool `true/false` in old code (`canCreateBranch` is string while `.root.CanCreateBranch` is bool)

Playing with string `"true"/"false"` is dangerous, so I think it time to always use bool. https://github.com/go-gitea/gitea/pull/23394/commits/99b961db9e7b8548ec14410b844fa9b39f9f8a66

Reply:
> Oh, bug.
> 
> I believe there was misuse between string `"true"/"false"` and bool `true/false` in old code (`canCreateBranch` is string while `.root.CanCreateBranch` is bool)
> 
> Playing with string `"true"/"false"` is dangerous, so I think it time to always use bool. [99b961d](https://github.com/go-gitea/gitea/commit/99b961db9e7b8548ec14410b844fa9b39f9f8a66)

Yep, agree.